node/tests/testEngine.go

```diff
@@ -0,0 +1,22 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/schema"
+)
+
+// Engine describes the interface that can be supplied to run tests on schemas.
+//
+// The PrototypeByName function can get its job done using only interface types
+// that we already know from outside any generated code, so you can write tests
+// that have no _compile time_ dependency on the generated code.  This makes it
+// easier for IDEs and suchlike to help you write and check the test functions.
+//
+// Ask for prototypes using the type name alone (no package prefix);
+// their representation prototypes can be obtained by appending ".Repr".
+type Engine interface {
```

Comment:
Just for discussion:

The other way I could imagine writing this would be more functionally: instead of this interface, which has to be held in a certain order, there could be a function like `Engine(t *testing.T, ts schema.TypeSystem) func(name string) ipld.NodePrototype`.

Usage would then involve holding onto that function returned, and using it the same places `engine.PrototypeByName` is used with this approach.

Already discussed this with @mvdan in another medium and he points out it's harder to add more methods in that approach.  Also true.

Reply:
Yep, and that would be akin to a constructor. I don't have a strong opinion on this, but I do find that "func that returns func with some state" can be a bit confusing, so I tend to avoid it whenever possible. Plus I think it's likely we'll extend this interface with time, perhaps to solve the problem involving repetition of test names etc.